Ficha9/grafo.c

```diff
@@ -78,26 +111,78 @@ int grafo_aresta(grafo *g, int origem, int dest)
 grafo* grafo_deLista(int* adjacencias, int n_arestas)
 {
     /* alinea 1.2 */
+    if(adjacencias == NULL || (n_arestas < 1)){
+        return NULL;
+    }
+
+    int i, tamanho = adjacencias[0];
+    for(int i = 1; i < 2*n_arestas; i++)

```

Comment:
why `2*`?

Reply:
Because the array that has the information on the connection between the vertices has double the space of the number of edges. For example, if you have 3 vertices, you can up to 6 edges (one to each direction), but the array will have 12 elements (ex: {0,1,1,0,1,2,2,1,0,2,2,0} ).